### PR TITLE
feat(dds): support dds instance set description and availability zone and slow log status

### PR DESCRIPTION
**Pull request by SabreKeyZ**
…og status

**What this PR does / why we need it**:
support dds instance set description and availability zone and slow log status
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
  will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
support dds instance set description and availability zone and slow log status
```

### CODE REVIEW

File: openstack/dds/v3/instances/requests.go [hunk 1]
```diff
@@ -56,6 +56,10 @@ type BackupPolicyOpts struct {
 	BackupPolicy BackupStrategy `json:"backup_policy" required:"true"`
 }
 
+type SlowLogStatus struct {
```
Comment (Jason-Zhang9309) [inline openstack/dds/v3/instances/requests.go:59]: this struct is never used

File: openstack/dds/v3/instances/urls.go [hunk 2]
```diff
@@ -29,3 +29,19 @@ func secondsLevelMonitoringURL(c *golangsdk.ServiceClient, instanceId string) st
...
+
+func availabilityZoneURL(c *golangsdk.ServiceClient, instanceId string) string {
+	return c.ServiceURL("instances", instanceId, "migrate")
+}
+
+func remarkURL(c *golangsdk.ServiceClient, instanceId string) string {
+	return c.ServiceURL("instances", instanceId, "remark")
+}
+
+func createStatusURL(c *golangsdk.ServiceClient, instanceId string, status string) string {
```
Comment (Jason-Zhang9309) [inline openstack/dds/v3/instances/urls.go:41]:
```suggestion
func updateSlowlogStatusURL(c *golangsdk.ServiceClient, instanceId string, status string) string {
```

File: openstack/dds/v3/instances/urls.go [hunk 3]
```diff
@@ -29,3 +29,19 @@ func secondsLevelMonitoringURL(c *golangsdk.ServiceClient, instanceId string) st
...
+	return c.ServiceURL("instances", instanceId, "slowlog-desensitization", status)
+}
+
+func getStatusURL(c *golangsdk.ServiceClient, instanceId string) string {
```
Comment (Jason-Zhang9309) [inline openstack/dds/v3/instances/urls.go:45]:
```suggestion
func getSlowlogStatusURL(c *golangsdk.ServiceClient, instanceId string) string {
```

File: openstack/dds/v3/instances/requests.go [hunk 4]
```diff
@@ -237,6 +241,66 @@ type EnabledOpts struct {
...
+// UpdateStatus is a method to update the slow log status.
+func UpdateStatus(c *golangsdk.ServiceClient, instanceId string, slowLogStatus string) error {
+	_, err := c.Put(createStatusURL(c, instanceId, slowLogStatus), nil, nil, &golangsdk.RequestOpts{
+		MoreHeaders: requestOpts.MoreHeaders,
+	})
+	return err
+}
+
+// GetStatus is a method to get the slow log status.
+func GetStatus(c *golangsdk.ServiceClient, instanceId string) (string, error) {
```
Comment (Jason-Zhang9309) [inline openstack/dds/v3/instances/requests.go:294]:
```suggestion
func GetSlowlogStatus(c *golangsdk.ServiceClient, instanceId string) (string, error) {
```

File: openstack/dds/v3/instances/requests.go [hunk 5]
```diff
@@ -237,6 +241,66 @@ type EnabledOpts struct {
...
+
+type DescriptionOpts struct {
+	Remark string `json:"remark" required:"true"`
+}
+
+// UpdateAvailabilityZone is a method to update the AvailabilityZone.
+func UpdateAvailabilityZone(c *golangsdk.ServiceClient, instanceId string, az string) (*AvailabilityZoneResp, error) {
+	opts := AvailabilityZoneOpts{
+		TargetAzs: az,
+	}
```
Comment (Jason-Zhang9309) [inline openstack/dds/v3/instances/requests.go:256]: it's better to this part in the provider

File: openstack/dds/v3/instances/requests.go [hunk 6]
```diff
@@ -237,6 +241,66 @@ type EnabledOpts struct {
...
+		MoreHeaders: requestOpts.MoreHeaders,
+	})
+	return &r, err
+}
+
+// UpdateRemark is a method to update the description.
+func UpdateRemark(c *golangsdk.ServiceClient, instanceId string, description string) error {
+	opts := DescriptionOpts{
+		Remark: description,
+	}
```
Comment (Jason-Zhang9309) [inline openstack/dds/v3/instances/requests.go:273]: it's better to this part in the provider

File: openstack/dds/v3/instances/requests.go (hunk 1 above)
Comment (SabreKeyZ) [reply openstack/dds/v3/instances/requests.go:59]: done

File: openstack/dds/v3/instances/urls.go (hunk 2 above)
Comment (SabreKeyZ) [reply openstack/dds/v3/instances/urls.go:41]: done

File: openstack/dds/v3/instances/urls.go (hunk 3 above)
Comment (SabreKeyZ) [reply openstack/dds/v3/instances/urls.go:45]: done

File: openstack/dds/v3/instances/requests.go (hunk 4 above)
Comment (SabreKeyZ) [reply openstack/dds/v3/instances/requests.go:294]: done

File: openstack/dds/v3/instances/requests.go [hunk 7]
```diff
@@ -237,6 +241,66 @@ type EnabledOpts struct {
...
+
+	_, err = c.Put(remarkURL(c, instanceId), b, nil, &golangsdk.RequestOpts{
+		MoreHeaders: requestOpts.MoreHeaders,
+	})
+	return err
+}
+
+// UpdateStatus is a method to update the slow log status.
+func UpdateStatus(c *golangsdk.ServiceClient, instanceId string, slowLogStatus string) error {
```
[the remark this answers is not in the document]
Comment (SabreKeyZ) [reply openstack/dds/v3/instances/requests.go:286]: done

File: openstack/dds/v3/instances/requests.go (hunk 6 above)
Comment (SabreKeyZ) [reply openstack/dds/v3/instances/requests.go:273]: done

File: openstack/dds/v3/instances/requests.go (hunk 5 above)
Comment (SabreKeyZ) [reply openstack/dds/v3/instances/requests.go:256]: done

File: openstack/dds/v3/instances/requests.go [hunk 8]
```diff
@@ -237,6 +237,66 @@ type EnabledOpts struct {
...
+
+type DescriptionOpts struct {
+	Remark string `json:"remark" required:"true"`
+}
+
+// UpdateAvailabilityZone is a method to update the AvailabilityZone.
+func UpdateAvailabilityZone(c *golangsdk.ServiceClient, instanceId string, az AvailabilityZoneOpts) (*AvailabilityZoneResp, error) {
+	opts := AvailabilityZoneOpts{
+		TargetAzs: az.TargetAzs,
+	}
```
Comment (Jason-Zhang9309) [inline openstack/dds/v3/instances/requests.go:252]:
```suggestion
func UpdateAvailabilityZone(c *golangsdk.ServiceClient, instanceId string, opts AvailabilityZoneOpts) (*AvailabilityZoneResp, error) {
```

**Pull request merged**